src/main/java/reposense/parser/CsvParser.java

```diff
@@ -76,6 +79,25 @@ public CsvParser(Path csvFilePath) throws IOException {
         return results;
     }
 
+    /**
+     * Returns the header of a CSV file, which is assumed to be the first non-empty / non-whitespace line in the file.
+     * The line is split into an array of Strings, using the comma symbol as delimiter.
+     *
+     * @throws IOException if the file has only empty or blank lines, or if there is an error accessing the file.
+     */
+    private String[] getHeader(BufferedReader reader) throws IOException {
+        String currentLine = "";
+
+        // read from file until we encounter a line that is neither blank nor empty
+        while (currentLine.length() == 0) {
+            currentLine = Optional.ofNullable(reader.readLine())
+                                  .map(String::trim)
+                                  .orElseThrow(IOException::new);
```

Comment:
I have fixed the line wrapping.

Reply:
@eugenepeh, I have made a change my approach for the exception handling.

I created a new exception, `InvalidCsvException`, which signifies that there is a major error in the contents of a CSV file, such that the CSV is totally unusable.
For example, the CSV file may be totally empty, or all of its records may be malformed.

Reason for creating new exception, rather than using `IOException`: The code becomes more readable as the exception is more specific.

There is a minor problem, though. Ideally, we would like the logger to inform the user if the `author-config.csv` and `group-config.csv` have major problems.

However, currently, RepoSense will silently ignore all `IOExceptions` that occur for `author-config.csv` and `group-config.csv`.

To fix this, we may need to throw the `InvalidCsvException` from `parse` in `CsvParser` to the main code in `RepoSense.java.`, where it can be handled for `author-config.csv` and `group-config.csv`.

Reply:
Sounds good to me